src/main/java/com/shrralis/ssdemo1/service/AuthServiceImpl.java

```diff
@@ -30,22 +36,63 @@
  */
 @Service
 public class AuthServiceImpl implements IAuthService {
-    @Resource
-    private UsersRepository repository;
+	private static final Logger logger = LoggerFactory.getLogger(AuthServiceImpl.class);
+	private static final String LOGIN_PATTERN = "^[A-Za-z_\\-.0-9]+$";
+	@Resource
+	private UsersRepository repository;
     @Autowired
     private PasswordEncoder passwordEncoder;
 
-    // TODO: make some checkings
     @Override
-    public JsonResponse signUp(RegisterUserDTO user) {
-        user.setPassword(passwordEncoder.encode(user.getPassword()));
-        System.out.println(user.getPassword());
-        return new JsonResponse(repository.save(User.Builder.anUser()
-                .setLogin(user.getLogin())
-                .setEmail(user.getEmail())
-                .setPassword(user.getPassword())
-                .setName(user.getName())
-                .setSurname(user.getSurname())
-                .build()));
+    public JsonResponse signUp(RegisterUserDTO user) throws AbstractShrralisException {
+	    if (user == null) {
+		    throw new NullPointerException("Passed `User` is null");
+	    }
+
+	    if (repository.getByLogin(user.getLogin()) != null) {
```

Comment:
I would recommend create and use method like isExistWithLogin/Email . There are no need to extract whole entity